src/app/content/constants.ts

```diff
@@ -18,3 +19,15 @@ export const highlightStyles: Array<{label: HighlightColorEnum, passive: string,
   {label: HighlightColorEnum.Purple, passive: '#cbcfff', focused: '#545ec8'},
   {label: HighlightColorEnum.Pink, passive: '#ffc5e1', focused: '#de017e'},
 ];
+
+const getBaseColors = (colors: {[key: string]: {[key: string]: string}}) => {
+  const bases: { [key: string]: string } = {};
+
+  Object.keys(colors).forEach((color) => {bases[color] = colors[color].base; });
+  return bases;
+};
+
+export const searchButtonStyles: {[key: string]: string} = {
```

Comment:
@TomWoodward what is the correct way to apply a custom type (in this case `SearchButtonColor` defined below) for object keys? I tried but ran into ts errors that I struggled to resolve

Reply:
the way you have it done here you'd just do 
```

Reply:
No, first I got the error `An index signature parameter type cannot be a union type. Consider using a mapped object type instead.` and then others as I tried to resolve it. But this code isn't needed anymore.

Reply:
did you try the mapped type as the error suggests? by doing: 
```
export const searchButtonStyles: {[key in SearchButtonColor]: string} = {
```